drivers/flash/Kconfig.disk

```diff
@@ -0,0 +1,13 @@
+# Disk flash emulator config
+
+# Copyright (c) 2023 Endress+Hauser AG
+# SPDX-License-Identifier: Apache-2.0
+
+config FLASH_DRIVER_DISK
+	bool "Use the flash on disk emulator"
```

Comment:
The convention for drivers is to define the Kconfig like so:

```
config FLASH_DRIVER_DISK
	default y
	depends on DT_HAS_ZEPHYR_DISK_FLASH_ENABLED
```

This way, the driver is enabled whenever the subsystem is available (`CONFIG_FLASH=y`) and an enabled node exists in the devicetree

Reply:
I didn't know this. So this would be the prefered way? Or is the select not necessary?
```
config FLASH_DRIVER_DISK
	bool "Use the flash on disk emulator"
	default y
	depends on DT_HAS_ZEPHYR_DISK_FLASH_ENABLED
	select FLASH_HAS_DRIVER_ENABLED
	help
	  This option enables the flash on disk emulator
	  which can be used to emulate a flash on a disk
	  device like sd-card or ramdisk.
```